File: 351001/Strizhak/rest_jpa/rest_jpa/Dtos/UserRequestTo.cs
using System.ComponentModel.DataAnnotations;
namespace rest_api.Dtos
{
    public class UserRequestTo
    {
        [Required]
        public long Id { get; set; }
        [Required]
        [StringLength(64, MinimumLength = 2)]
        public string Login { get; set; } = null!;

        [Required]
        [StringLength(128, MinimumLength = 8)]
        public string Password { get; set; } = null!;

        [Required]
        [StringLength(64, MinimumLength = 2)]
        public string Firstname { get; set; } = null!;

        [Required]
        [StringLength(64, MinimumLength = 2)]
        public string Lastname { get; set; } = null!;
    }
}

File: 351004/Mukhin/Lab1/restApi/publisher/src/main/java/com/example/restApi/dto/request/UserRequestTo.java
package com.example.restApi.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public class UserRequestTo {

    @JsonProperty("login")
    @NotBlank(message = "Login cannot be blank")
    @Size(min = 2, max = 64, message = "Login must be between 2 and 64 characters")
    private String login;

    @JsonProperty("password")
    @NotBlank(message = "Password cannot be blank")
    @Size(min = 8, max = 128, message = "Password must be between 8 and 128 characters")
    private String password;

    @JsonProperty("firstname")
    @NotBlank(message = "Firstname cannot be blank")
    @Size(min = 2, max = 64, message = "Firstname must be between 2 and 64 characters")
    private String firstname;

    @JsonProperty("lastname")
    @NotBlank(message = "Lastname cannot be blank")
    @Size(min = 2, max = 64, message = "Lastname must be between 2 and 64 characters")
    private String lastname;

    public String getLogin() {
        return login;
    }

    public void setLogin(String login) {
        this.login = login;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getFirstname() {
        return firstname;
    }

    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    public String getLastname() {
        return lastname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    private String role;

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }
}